packages/auth-core-plugins/src/AuthPluginParent.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+import { LoginOptions } from '@deephaven/jsapi-types';
+import {
+  LOGIN_OPTIONS_REQUEST,
+  requestParentResponse,
+} from '@deephaven/jsapi-utils';
+import Log from '@deephaven/log';
+import { LoadingOverlay } from '@deephaven/components';
+import { AuthPlugin, AuthPluginProps } from '@deephaven/auth-plugin';
+
+const log = Log.module('AuthPluginParent');
+
+function getWindowAuthProvider(): string {
+  return new URLSearchParams(window.location.search).get('authProvider') ?? '';
+}
+
+/**
+ * AuthPlugin that tries to delegate to the parent window for authentication. Fails if there is no parent window.
+ */
+function Component({
+  client,
+  onSuccess,
+  onFailure,
+}: AuthPluginProps): JSX.Element {
+  const [error, setError] = useState<unknown>();
+  useEffect(() => {
+    async function login() {
+      try {
+        log.info('Logging in by delegating to parent window...');
+        const loginOptions = await requestParentResponse<LoginOptions>(
+          LOGIN_OPTIONS_REQUEST
+        );
+
+        await client.login(loginOptions);
```

Comment:
Would this work with something like if the parent required oauth or saml? Or just not something we currently support in DHC? 

What would happen if the parent wanted to show a login screen?

Reply:
The parent authentication is retrieving the auth details from the parent - we don't have any option to show a login here in this window and I think that's alright, as the parent should already be authenticated and just passing the token. If we do need to show a login screen, we'll need to have another value returned from `requestParentResponse`.

As for showing a login screen in general, Colin has made an oidc plugin as well to test it out with keycloak: https://github.com/deephaven/deephaven-js-plugins/compare/main...niloc132:deephaven-js-plugins:draft-keycloak-auth-plugin